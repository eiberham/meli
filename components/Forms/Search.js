import React from 'react';
import { Input } from 'semantic-ui-react';

const Search = props => {
    const {
        values: { search },
        errors,
        touched,
        handleChange,
        handleSubmit,
        isValid,
        setFieldTouched
    } = props;

    const change = (name, e) => {
        e.persist();
        handleChange(e);
        setFieldTouched(name, true, false);
    };

    return (
        <React.Fragment>
            <form onSubmit={handleSubmit}>
                <Input
                    id="search"
                    name="search"
                    type="text"
                    size="massive"
                    placeholder="Nunca dejes de buscar ..."
                    value={term}
                    onChange={change.bind(null, "url")}
                    action={{
                        placeholder: '',
                        color: 'teal',
                        content: 'Search',
                        onClick: () => {
                            
                        }
                    }}
                />
                <p>{ term }</p>
            </form>
        </React.Fragment>
    );
}

export default Search;