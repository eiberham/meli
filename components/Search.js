import React, { useState, useRef } from "react";
import { Formik } from 'formik';
import * as Yup from 'yup';
import { Input } from "semantic-ui-react";

const Search = () => {
    const [term, setTerm] = useState('');
    return (
        <React.Fragment>
            <Input
                id="search"
                name="search"
                type="text"
                size="massive"
                placeholder="Nunca dejes de buscar ..."
                value={term}
                onChange={(e) => setTerm(e.target.value)}
                action={{
                    placeholder: '',
                    color: 'teal',
                    content: 'Search',
                    onClick: () => {
                        
                    }
                }}
            />
            <p>{ term }</p>
        </React.Fragment>
    )
};

export default Search;