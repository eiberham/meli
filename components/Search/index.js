import React, { useState, useRef } from "react";
import { Formik } from 'formik';
import SearchForm from "../Forms/SearchForm";
import Router from 'next/router';
import { withRouter } from 'next/router';

const Search = props => {
    // Save in context the search value
    const { router } = props;
    return (
        <React.Fragment>
            <Formik
                render={props => <SearchForm {...props} />}
                initialValues={{ search: '' }}
                onSubmit={ ({search}, actions) => {
                    try {
                        // Hacer la request con axios o fetch
                        router.push(`/items?search=${search}`);
                    }catch(error){
                        console.error(error);
                    }

                }}
            />
        </React.Fragment>
    )
};

export default withRouter(Search);