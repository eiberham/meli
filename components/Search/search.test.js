import React from 'react';
import mockAxios from 'axios';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Search from './index';

describe('Search', () => {
    it('matches snapshot', () => {
        const component = renderer.create(<Search />)
        .toJSON();
        expect(component).toMatchSnapshot();
    })

    it('send a request to the right api and append the search parameter', () => {
        const wrapper = shallow(<Search />);
        const input = wrapper.find('input')
        /* input.instance().value = "musculosas"; */
        input.value = "musculosas";

        /* jest.mock('axios') */

        mockAxios.get.mockImplementationOnce(() =>
            Promise.resolve({
                data: {
                    results: ["cute.jpg"]
                }
            })
        );

        /* axios.get.mockImplementation(async () => response) */

        axios.get.mockResolvedValue(response);

        /* input.trigger('keyup.enter') */
        input.simulate('keyDown', {
            which: 27,
            target: {
              blur() {
                input.simulate('blur');
              },
            },
        });

        expect(mockAxios.get).toHaveBeenCalledWith(`http://localhost:3000/api/items?search=musculosas`)
        expect(mockAxios.get).toHaveBeenCalledTimes(1);
    })
})