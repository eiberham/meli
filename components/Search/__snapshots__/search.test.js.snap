// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search matches snapshot 1`] = `
<div
  className="search"
>
  <form
    className="css-1vr8bhw"
    onSubmit={[Function]}
  >
    <div
      className="ui small fluid action input"
    >
      <input
        id="search"
        name="search"
        onChange={[Function]}
        placeholder="Nunca dejes de buscar ..."
        type="text"
        value=""
      />
      <button
        className="ui icon button"
        onClick={[Function]}
        placeholder=""
      >
        <i
          aria-hidden="true"
          className="search icon"
          onClick={[Function]}
        />
      </button>
    </div>
  </form>
</div>
`;
